projects/igniteui-angular/src/lib/services/exporter-common/base-export-service.ts

```diff
@@ -318,15 +325,174 @@ export abstract class IgxBaseExporter {
+            }
 
-        if (tagName === 'igx-grid') {
-            this.prepareGridData(grid as IgxGridComponent, options, hasFiltering, hasSorting);
-        } if (tagName === 'igx-tree-grid') {
-            this.prepareTreeGridData(grid as IgxTreeGridComponent, options, hasFiltering, hasSorting);
+            if (tagName === 'igx-tree-grid') {
+                this.prepareTreeGridData(grid as IgxTreeGridComponent, hasFiltering, hasSorting);
+            }
         }
     }
 
-    private prepareGridData(grid: IgxGridComponent, options: IgxExporterOptionsBase, hasFiltering: boolean, hasSorting: boolean) {
+    private prepareHierarchicalGridData(grid: IgxHierarchicalGridComponent, hasFiltering: boolean, hasSorting: boolean) {
+
+        const skipOperations =
+            (!hasFiltering || !this.options.ignoreFiltering) &&
+            (!hasSorting || !this.options.ignoreSorting);
+
+        if (skipOperations) {
+            const data = grid.filteredSortedData;
+            this.addHierarchicalGridData(grid, data);
+        } else {
+            let data = grid.data;
+
+            if (hasFiltering && !this.options.ignoreFiltering) {
+                const filteringState: IFilteringState = {
+                    expressionsTree: grid.filteringExpressionsTree,
+                    advancedExpressionsTree: grid.advancedFilteringExpressionsTree,
+                };
+
+                filteringState.strategy = grid.filterStrategy;
+
+                data = DataUtil.filter(data, filteringState, grid);
+            }
+
+            if (hasSorting && !this.options.ignoreSorting) {
+                this._sort = cloneValue(grid.sortingExpressions[0]);
+
+                data = DataUtil.treeGridSort(data, grid.sortingExpressions, grid.sortStrategy);
+            }
+
+            this.addHierarchicalGridData(grid, data);
+        }
+    }
+
+    private addHierarchicalGridData(grid: IgxHierarchicalGridComponent, records: any[]) {
+        const childLayoutKeys = grid.childLayoutKeys;
+        const childGrids = grid.hgridAPI.getChildGrids(true);
+        const columnFields = this._ownersMap.get(grid).columns.map(col => col.field);
+
+        for(const entry of records) {
+            const expansionStateVal = grid.expansionStates.has(entry) ? grid.expansionStates.get(entry) : false;
+
+            const dataWithoutChildren = Object.keys(entry)
+                .filter(k => columnFields.includes(k))
+                .reduce((obj, key) => {
+                    obj[key] = entry[key];
+                    return obj;
+                }, {});
+
+            const hierarchicalGridRecord: IExportRecord = {
+                data: dataWithoutChildren,
+                level: 0,
+                type: ExportRecordType.HierarchicalGridRecord,
+                owner: grid,
+            };
+
+            this.flatRecords.push(hierarchicalGridRecord);
+
+            for (const key of childLayoutKeys) {
```

Comment:
Why not for of `childLayoutList`?
```suggestion
            for (const island of childLayoutList) {
```